src/modules/emit_sound.rs

```diff
@@ -120,34 +121,36 @@ fn emit_sound(marker: MainThreadMarker, sound: String, channel: i32) {
 }
 
 fn emit_sound_full(marker: MainThreadMarker, info: SoundInfo) {
-    if let Some(player) = unsafe { engine::player_edict(marker) } {
-        let entity = unsafe { player.as_ptr().add(info.from as usize).as_mut() };
+    let entity = unsafe {
+        (*engine::sv_edicts.get(marker))
+            .add(info.from as usize)
+            .as_mut()
+    };
+
+    if entity.is_none() {
```

Comment:
This check does nothing. You're adding something to a pointer, it will never be NULL. You need to check num_edicts as I mentioned before.